Objects/typevarobject.c

```diff
@@ -458,6 +461,73 @@ unpack_typevartuples(PyObject *params)
     }
 }
 
+static PyObject *
+typeparam_reduce_anonymous(PyObject *self, PyObject *owner)
+{
+    assert(PyTuple_Check(owner));
+    assert(PyTuple_GET_SIZE(owner) == 3);
+    PyObject *module_name = PyTuple_GET_ITEM(owner, 0);
+    PyObject *qualname = PyTuple_GET_ITEM(owner, 1);
+    PyObject *index = PyTuple_GET_ITEM(owner, 2);
+    assert(PyUnicode_Check(module_name));
+    assert(PyUnicode_Check(qualname));
+    assert(PyLong_Check(index));
+    PyObject *ret = NULL;
+    PyObject *typing = NULL;
+    PyObject *restore_func = NULL;
+    PyObject *module = NULL;
+    PyObject *path = NULL;
+    PyObject *obj = NULL;
+    PyObject *args = NULL;
+
+    typing = PyImport_ImportModule("_typing");
+    if (typing == NULL) {
+        goto done;
+    }
+    restore_func = PyObject_GetAttrString(typing, "_restore_anonymous_typeparam");
+    if (restore_func == NULL) {
+        goto done;
+    }
+
+    module = PyImport_Import(module_name);
+    if (module == NULL) {
+        goto done;
+    }
+    path = PyUnicode_Split(qualname, _Py_LATIN1_CHR('.'), -1);
+    if (path == NULL) {
+        goto done;
+    }
+    if (PyList_GET_SIZE(path) == 0) {
+        PyErr_SetString(PyExc_ValueError, "invalid typeparam owner qualname");
+        goto done;
+    }
+    obj = module;
+    Py_INCREF(obj);
+    for (Py_ssize_t i = 0; i < PyList_GET_SIZE(path); i++) {
```

Comment:
At first glance, I think we can store `PyList_GET_SIZE` separately without to recompute it at each loop iteration. I'm not entirely sure about it as `PyObject_GetAttr` may call arbitrary code, but I'm not sure we can actually affect `path` even with side-effects as no Python code can actually affect it (we're not using some call to `.split()`, we're just assuming that `path` is a string and use its buffer directly).

However, since I'm not really sure about this, we can leave it as is (this is just in case we want to look for micro-optimizations)

Reply:
Its how I see it done all over the codebase so likewise I just did it this way. `path` will be mostly a single element or very small anyway.